bundles/org.openhab.binding.miele/src/main/resources/OH-INF/addon/addon.xml

```diff
@@ -8,4 +8,11 @@
 	<description>This is the binding for Miele@home appliances</description>
 	<connection>local</connection>
 
+	<discovery-methods>
+		<discovery-method>
+			<service-type>mdns</service-type>
+			<mdns-service-type>_mieleathome._tcp.local.</mdns-service-type>
+		</discovery-method>
+	</discovery-methods>
```

Comment:
This will correctly identify the XGW 3000 gateway used for the old local Zigbee-based Miele@home, i.e. the **miele** binding. So looks good:

![image](https://github.com/openhab/openhab-addons/assets/19519842/01b299c6-bc2e-42dd-acd6-3f39d4022199)

However, it will also identify Miele Wi-Fi devices, such as my dishwasher:

![image](https://github.com/openhab/openhab-addons/assets/19519842/984d137d-b481-4592-b9f2-9088f870e7ef)

Now comes the tricky part: Newer Wi-Fi devices are usually integrated through Miele Cloud. But XGW 3000 is also integrated with Miele Cloud so that the Miele app is compatible with both Wi-Fi and Zigbee communication modules.

This means that the **mielecloud** binding can be used to integrate all my appliances - Zigbee/XGW 3000 as well as Wi-Fi/cloud. And the "miele" binding can also be used to integrate all my appliances, also the Wi-Fi/cloud-based ones.

Because of this, I think the best we can do is to suggest both **miele** and **mielecloud** based on the same discovery-method definition. Unless the name "Miele XGW3000" can be matched as well? In this case this could be used to match **miele** and everything else could be used to match **mielecloud**. That would mean that if **only** XGW 3000 is found, but no other devices, **mielecloud** will not be suggested. I think this is okay, since users not having **any** Wi-Fi devices would probably prefer only the local integration anyway.

Reply:
Umm.

IMHO I think we should suggest both addons when more that one might apply. I did this already for 'hpprinter' and 'ipp' printer, and for MediaRenders like Sony, Yamaha, LG etc. which may both recommended the manufacturer specific addons and also the generic `upnpcontrol` addon.

But in the Miele case we can probably be more specific
- service type `_mieleathome._tcp.local.` causes `mielecloud` to (always) be suggested
- service type `_mieleathome._tcp.local.` combined with `path=\Rest` causes `miele` to be suggested too

Or ??

Reply:
Is it possible to match by the absence of "/rest/"? That would be even better for matching **mielecloud**.

Reply:
> Is it possible to match by the absence of "/Rest/"?

The complete absence of a 'path' property would cause the pattern matcher to fail (i.e. 'null' property). But fortunately your two above mentioned mDNS results have either 'path=\' or 'path=\Rest\', so we could probably discern on that. For example as follows..

**Cloud**
```
	<discovery-methods>
		<discovery-method>
			<service-type>mdns</service-type>
			<mdns-service-type>_mieleathome._tcp.local.</mdns-service-type>
			<match-properties>
				<match-property>
					<name>path</name>
					<regex>\/</regex>
				</match-property>
			</match-properties>
		</discovery-method>
	</discovery-methods>
```

**Gateway**
```
	<discovery-methods>
		<discovery-method>
			<service-type>mdns</service-type>
			<mdns-service-type>_mieleathome._tcp.local.</mdns-service-type>
			<match-properties>
				<match-property>
					<name>path</name>
					<regex>\/Rest\/</regex>
				</match-property>
			</match-properties>
		</discovery-method>
	</discovery-methods>
```